control-plane/subcommand/sync-catalog/command.go

```diff
@@ -523,11 +525,38 @@ func (c *Command) recordMetrics() (*prometheus.PrometheusSink, error) {
 	return sink, nil
 }
 
+func (c *Command) validateToken(token string, consulConfig *consul.Config) bool {
+	if token == "" {
+		return false
+	}
+
+	// Create a new consul client.
+	consulClient, err := consul.NewClientFromConnMgr(consulConfig, c.connMgr)
+	if err != nil {
+		c.logger.Error("failed to create Consul API client", "err", err)
+		return false
+	}
+
+	tok, _, err := consulClient.ACL().TokenReadSelf(&api.QueryOptions{Token: token})
+	if err != nil && tok != nil {
+		c.logger.Error("failed to validate ACL token", "err", err)
+		return false
+	}
+
+	spew.Dump(tok)
+	c.logger.Info("ACL token validated", "token", tok)
+
+	return true
+}
+
 // authorizeMiddleware validates the token and returns http handler.
-func (c *Command) authorizeMiddleware() func(http.Handler) http.Handler {
+func (c *Command) authorizeMiddleware(consulConfig *consul.Config) func(http.Handler) http.Handler {
+
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			// TO-DO: Validate the token and proceed to the next handler
+			token := r.Header.Get("X-Consul-Token")
+			c.validateToken(token, consulConfig)
```

Comment:
assuming this would be something like

```go
if !c.validateToken(token, consulConfig) {
    http.Error(w, 500, "invalid token")
    return
}
next.ServeHTTP(w, r)
```